autodiff/reverse/var/var.hpp

```diff
@@ -1091,42 +1374,78 @@ struct Variable
-template<typename T> bool operator<=(const Variable<T>& l, const ExprPtr<T>& r) { return l.expr->val <= r->val; }
-template<typename T> bool operator>=(const Variable<T>& l, const ExprPtr<T>& r) { return l.expr->val >= r->val; }
-template<typename T> bool operator<(const Variable<T>& l, const ExprPtr<T>& r) { return l.expr->val < r->val; }
-template<typename T> bool operator>(const Variable<T>& l, const ExprPtr<T>& r) { return l.expr->val > r->val; }
-
-template<typename T> bool operator==(const ExprPtr<T>& l, const Variable<T>& r) { return l->val == r.expr->val; }
-template<typename T> bool operator!=(const ExprPtr<T>& l, const Variable<T>& r) { return l->val != r.expr->val; }
-template<typename T> bool operator<=(const ExprPtr<T>& l, const Variable<T>& r) { return l->val <= r.expr->val; }
-template<typename T> bool operator>=(const ExprPtr<T>& l, const Variable<T>& r) { return l->val >= r.expr->val; }
-template<typename T> bool operator<(const ExprPtr<T>& l, const Variable<T>& r) { return l->val < r.expr->val; }
-template<typename T> bool operator>(const ExprPtr<T>& l, const Variable<T>& r) { return l->val > r.expr->val; }
-
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator==(const U& l, const Variable<T>& r) { return l == r.expr; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator!=(const U& l, const Variable<T>& r) { return l != r.expr; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator<=(const U& l, const Variable<T>& r) { return l <= r.expr; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator>=(const U& l, const Variable<T>& r) { return l >= r.expr; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator<(const U& l, const Variable<T>& r) { return l < r.expr; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator>(const U& l, const Variable<T>& r) { return l > r.expr; }
-
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator==(const Variable<T>& l, const U& r) { return l.expr == r; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator!=(const Variable<T>& l, const U& r) { return l.expr != r; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator<=(const Variable<T>& l, const U& r) { return l.expr <= r; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator>=(const Variable<T>& l, const U& r) { return l.expr >= r; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator<(const Variable<T>& l, const U& r) { return l.expr < r; }
-template<typename T, typename U, EnableIf<isArithmetic<U>>...> bool operator>(const Variable<T>& l, const U& r) { return l.expr > r; }
+
+struct ExprValue {
+  template<typename T, EnableIf<isArithmetic<T>>...> static auto value(const T& t) { return t; }
```

Comment:
Similar to my comments below for `coerce`, we can have these `value` methods as free functions.